tests/test_api.py

```diff
@@ -143,14 +143,16 @@ def sendall(self, *args, **kwargs):
 
 class MockRestApiServer(RestApiServer):
 
-    def __init__(self, Handler, request, config=None):
+    def __init__(self, Handler, request, config=None, patroni=None):
         self.socket = 0
         self.serve_forever = Mock()
+        if patroni is None:
+            patroni = MockPatroni()
         MockRestApiServer._BaseServer__is_shut_down = Mock()
         MockRestApiServer._BaseServer__shutdown_request = True
         config = config or {'listen': '127.0.0.1:8008', 'auth': 'test:test', 'certfile': 'dumb', 'verify_client': 'a',
                             'http_extra_headers': {'foo': 'bar'}, 'https_extra_headers': {'foo': 'sbar'}}
-        super(MockRestApiServer, self).__init__(MockPatroni(), config)
+        super(MockRestApiServer, self).__init__(patroni, config)
```

Comment:
IMO, this refactoring is not necessary. Please just add the necessary tags to the `MockPatroni.tags`, they are harmless for remaining tests.